File: src/main/java/edu/dcccd/trans/entity/RegistrationForm.java
package edu.dcccd.trans.entity;

import lombok.Data;
import org.springframework.security.crypto.password.PasswordEncoder;

@Data
public class RegistrationForm {

  private String username;
  private String password;
  private String confirm;

  public User toUser( PasswordEncoder passwordEncoder) {
    return new User(username, passwordEncoder.encode(password));
  }

  public String getUsername() {
    return username;
  }

  public String getPassword() {
    return password;
  }

  public String getConfirm() {
    return confirm;
  }
}

File: src/main/kotlin/edu/dcccd/trans/entity/RegistrationForm.kt
package edu.dcccd.trans.entity

import org.springframework.security.crypto.password.PasswordEncoder

data class RegistrationForm(var username: String? = null,
                            var password: String? = null,
                            var confirm: String? = null
)
{
    fun toUser(passwordEncoder: PasswordEncoder): User {
        return User(null, this.username!!, passwordEncoder.encode(this.password))
    }
}
